shuup/front/apps/product_filter/views.py

```diff
@@ -0,0 +1,266 @@
+                        attribute=attribute,
+                        defaults={
+                            'enabled': enabled,
+                            'shop': get_shop(self.request)
+                        }
+                    )
+            elif '-' in key:
+                b_attr_name = key.split('-')[-1]
+                b_enabled = data['enabled-%s' % (b_attr_name)]
+                b_filter_obj = BasicAttributesFilterSettingsModel.objects\
+                    .update_or_create(
+                        attribute_name=b_attr_name,
+                        defaults={
+                            'enabled': b_enabled,
+                            'shop': get_shop(self.request)
+                        }
+                    )
+
+
+class ProductFilterSettingsView(FormPartsViewMixin, SaveFormPartsMixin,
+                                CreateOrUpdateView):
+    form_class = None
+    template_name = 'shuup/product_filter/admin/edit.jinja'
+    context_object_name = 'product_filter'
+    base_form_part_classes = [
+        BaseFilterSettingsFormPart,
+        CategoriesFilterSettingsFormPart,
+        AttributesFilterSettingsFormPart,
+    ]
+
+    def get_object(self):
+        try:
+            instance = BasicFilterSettingsModel.objects.get(
+                shop=get_shop(self.request)
+            )
+        except:
+            instance = BasicFilterSettingsModel()
+        return instance
+
+    @atomic
+    def form_valid(self, form):
+        return self.save_form_parts(form)
+
+    def get_success_url(self):
+        return reverse('shuup_admin:product_filter.list')
+
+
+class ProductFilterView(ListView):
+    form_class = ProductListForm
+    template_name = 'shuup/product_filter/front/product_filter.jinja'
+    model = Product
+    context_object_name = 'products'
+
+    def dispatch(self, request, *args, **kwargs):
+        self.form = ProductListForm(
+            request=self.request,
+            shop=self.request.shop,
+            category=None,
+            data=self.request.GET
+        )
+        return super(ProductFilterView, self).dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        if not self.form.is_valid():
+            return Product.objects.none()
+        data = self.form.cleaned_data
+        if not data:  # pragma: no cover
+            return Product.objects.none()
+        products = Product.objects.filter(get_query_filters(self.request, None, data=data))
+        return get_product_queryset(products, self.request, None, data)
+
+    def get_context_data(self, **kwargs):
+        context = super(ProductFilterView, self).get_context_data(**kwargs)
+        context['filter_form'] = self.form
+        context['layout'] = 'vertical'
+        del self.form.fields['sort']
+        del self.form.fields['q']
+
+        try:
+            settings = BasicFilterSettingsModel.objects.get(
+                shop=get_shop(self.request),
+            )
+            print(settings.layout)
```

Comment:
☝️

Reply:
For these run `flake8`, `isort` and check the license headers for new files. You can also find this info from those failed Travis jobs.